src/component/api-token/ApiTokenCreate/ApiTokenCreate.tsx

```diff
@@ -0,0 +1,184 @@
+import { styles as commonStyles } from '../../../component/common';
+import { IApiTokenCreate } from '../../../hooks/api/actions/useApiTokensApi/useApiTokensApi';
+import useEnvironments from '../../../hooks/api/getters/useEnvironments/useEnvironments';
+import useProjects from '../../../hooks/api/getters/useProjects/useProjects';
+import Dialogue from '../../common/Dialogue';
+import MySelect from '../../common/select';
+import { useStyles } from './styles';
+
+const GLOBAL = ':global:';
+const ALL = '*';
+const TYPE_ADMIN = 'ADMIN';
+const TYPE_CLIENT = 'CLIENT';
+
+interface IApiTokenCreateProps {
+    showDialog: boolean;
+    closeDialog: () => void;
+    createToken: (token: IApiTokenCreate) => Promise<void>;
+}
+
+interface IDataError {
+    username?: string;
+    general?: string;
+}
+
+const INITIAL_DATA: IApiTokenCreate = {
+    username: '',
+    type: TYPE_CLIENT,
+    project: ALL,
+    environment: ALL
+}
+
+const ApiTokenCreate = ({
+    showDialog,
+    closeDialog,
+    createToken,
+}: IApiTokenCreateProps) => {
+    const styles = useStyles();
+    const [data, setData] = useState(INITIAL_DATA);
+    const [error, setError] = useState<IDataError>({});
+    const { projects } = useProjects();
+    const { environments } = useEnvironments();
+
+    const clear = () => {
+        setData({...INITIAL_DATA});
+        setError({});
+    }
+
+    const onCancel = (e: Event) => {
+        clear();
+        closeDialog();
+    };
+
+    const isValid = () => {
+        if(!data.username) {
+            setError({username: 'Username is required.'});
+            return false;
+        } else {
+            setError({})
+            return true;
+        }
+    }
+
+    
+
+    const submit = async () => {
+        if(!isValid()) {
+            return;
+        }
+        
+        try {
+            await createToken(data);
+            clear();
+            closeDialog();
+        } catch (error) {
+            setError({general: 'Unable to create new API token'});
+        }
+
+    }
+
+    const setType = (event: React.ChangeEvent<{value: string }>) => {
+        const value = event.target.value;
+        if(value === TYPE_ADMIN) {  
+            setData({...data, type: value, environment: ALL, project: ALL})
+
+        } else {
+            setData({...data, type: value})
+        }
+        
+    }
+
+    const setUsername = (event: React.ChangeEvent<{value: string }>) => {
+        const value = event.target.value;
+        setData({...data, username: value})
+    }
+
+    const setProject = (event:  React.ChangeEvent<{value: string }>) => {
+        const value = event.target.value;
+        setData({...data, project: value})
+    }
+
+    const setEnvironment = (event: React.ChangeEvent<{value: string }>) => {
+        const value = event.target.value;
+        setData({...data, environment: value})
+    }
+
+    const selectableProjects = [{id: '*', name: 'ALL'}, ...(projects || [])].map(i => ({
```

Comment:
Are these necessary? Both hooks return an empty array if there is no data.

Reply:
A of course, did not think about what the hook actually returned. 

They will also slightly change now that you cannot select  "Any"  environment.